### PR TITLE
Export accounts as JSON or CSV

### LINKED ISSUE #2147: Account backup functionality needs to be available
- allow the user to backup/export his accounts

(Split from https://github.com/ethcore/parity/issues/2111)

### PR DESCRIPTION
**Pull request by ngotchac**
Closes #2147 

Enables the user to export its Accounts to CSV or JSON files.

### CODE REVIEW

**Comment by jacogr** [comment]
Couple of comments -

1. Would probably just have the `.json` option since it should be more 'portable' between implementations
2. We need a restore option as well when we have the backup
3. Are we sure this is not better suited backed with `personal_backupAccounts` & `personal_restoreAccounts` APIs? @gavofyork ?

**Comment by gavofyork** [comment]
what would those APIs do? i think it's probably fine to do it in the browser from what i can see

**Comment by derhuerst** [comment]
> Would probably just have the .json option since it should be more 'portable' between implementations

👍 JSON feels a lot more robust than CSV. Beginners won't care what format it is and advanced users may fiddle with both.

**Comment by ngotchac** [comment]
I guess you're right about the CSV file... I was just thinking of users who uses Excel a lot who wants to copy/paste easily a vector of addresses...

**Comment by ngotchac** [comment]
Shouldn't the import be in another issue ? It needs more work (nice file upload UI at least), and would only be available for the Address Book without full export of accounts (keys)

**Comment by ngotchac** [comment]
Import logged here : #2885

File: js/src/ui/Actionbar/Export/export.js [hunk 1]
```diff
@@ -38,19 +39,45 @@ class ActionbarExport extends Component {
...
+
+    switch (filetype) {
+      case 'json':
+        return JSON.stringify(data, null, 4);
+      default:
+        return data.toString();
+    }
+  }
+
+  handleExport = () => {
```
Comment (derhuerst) [inline js/src/ui/Actionbar/Export/export.js]: is this abstraction worth it? also `getExtension` feels like we're not going to need it for new.

File: js/src/ui/Actionbar/Export/export.js [hunk 2]
```diff
@@ -38,19 +39,45 @@ class ActionbarExport extends Component {
...
+  onDownloadBackup = (filetype) => {
     const { filename, content } = this.props;
 
-    const text = (typeof content === 'string')
-      ? content
-      : JSON.stringify(content, null, 4);
+    const text = this.contentAsString(content, filetype);
+    const extension = this.getExtension(filetype);
 
     const blob = new Blob([ text ], { type: 'text/plain;charset=utf-8' });
```
Comment (derhuerst) [inline js/src/ui/Actionbar/Export/export.js]: shouldn't the mime type corresponding to `filetype` be passed as `type`?

**Comment by ngotchac** [comment]
@derhuerst Comments fixed!

**Comment by jacogr** [comment]
Happy to have imports seperated, just would like to keep this back a bit until that is ready - since it will be first thing people ask for.

**Comment by gavofyork** [comment]
onice until there are imports.

**Comment by gavofyork** [comment]
are we doing import functionality too or is this going to stay onice forever?

**Comment by ngotchac** [comment]
Yep it's coming. Should be really easy after #3279 is merged (new Import component)

**Comment by ngotchac** [comment]
Yep ; should be easy after #3279 gets merged

**Comment by gavofyork** [comment]
was merged 5 days ago... :-)

**Comment by ngotchac** [comment]
#3433  (And it was 3 days ago to be fair, during a week-end ;) )

**Pull request merged**